### ISSUE TITLE
File System Disks Bug

### ISSUE THREAD

**Issue by Noorullah-Ahmadzai** [linuxdeepin/dde-file-manager#307]
## Disk Bug
This bug makes my computer very slow.
I only have 3 disks. The system disk is ext4 and the remaining 2 is NTFS. The rest of them is randomly added by the file system.

![image](https://user-images.githubusercontent.com/28804764/99906207-7108a880-2cf3-11eb-96d0-b45d1b593019.png)

Here is how a proof in another file system where everything looks normal.

![image](https://user-images.githubusercontent.com/28804764/99906277-cb096e00-2cf3-11eb-9175-f26077db1d7e.png)

**Comment by vishnu-dev** [comment]
Just to mention, those SQUASH filesystems are added when you install a snap. Each snap is contained in a SQUASH filesystem. Although it is a bug that deepin has to resolve.

**Comment by Noorullah-Ahmadzai** [comment]
That means I have to remove all my snaps which is not a realistic solution. 

I hate this bug, it makes the file system very slow. 🐌

**Comment by vishnu-dev** [comment]
@xmuli Can this issue get more attention? This issue is very annoying as it slows down the file manager and sometimes crashes it.

**Comment by itsXuSt** [comment]
Hi @Noorullah-Ahmadzai , for **temporary** resolution, you can append some lines in file '/usr/lib/udev/rules.d/80-udisks2.rules' **or** '/etc/udev/rules.d/80-udisks2.rules' if the second file exists.
`KERNEL == "loop *", ENV {UDISKS_IGNORE} = "1"`
and then reboot your system. then the icons of squash fs gone.

**Comment by Noorullah-Ahmadzai** [comment]
@itsXuSt I tried it, No Luck after the reboot. :{

**Comment by itsXuSt** [comment]
> @itsXuSt I tried it, No Luck after the reboot. :{

ummm, should be useful, this resolution is provided by deepin users of deepin bbs.
https://bbs.deepin.org/post/203517
see this for reference.
could you please upload a snapshot of the property page of the extra icons (those squashfs partitions)? wanna see its' device descriptor.

**Comment by dandiws** [comment]
Thanks @itsXuSt , it works after I removed the spaces: `KERNEL=="loop*", ENV{UDISKS_IGNORE}="1"`

**Comment by itsXuSt** [comment]
> Thanks @itsXuSt , it works after I removed the spaces: `KERNEL=="loop*", ENV{UDISKS_IGNORE}="1"`

u r welcome, very happy it is useful for u.

**Comment by foresee-io** [comment]
The problem also appears when using [xDroid](https://www.linzhuotech.com/Product/download). See [this post](https://bbs.deepin.org/zh/post/215146)(Chinese) in deepin forum.

**Comment by itsXuSt** [comment]
> The problem also appears when using [xDroid](https://www.linzhuotech.com/Product/download). See [this post](https://bbs.deepin.org/zh/post/215146)(Chinese) in deepin forum.

yes, xDroid will create virual partition when install applications. file manager did not filter these loop devices.
and it can be solved by using this solution too.

**Comment by Noorullah-Ahmadzai** [comment]
>  Thanks @itsXuSt , it works after I removed the spaces: `KERNEL=="loop*", ENV{UDISKS_IGNORE}="1"`

Can't believe it actually worked!
Thank You, Guys!

I'm gonna keep the issue open until it's shipped by default.
